File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/job/JobAllowMultiCraftRequestSetMessage.scala





















// Generated on 05/03/2013 10:34:44
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait JobAllowMultiCraftRequestSetMessage extends MessageDeserializer {
  val messageId = 5749

  protected trait Message extends BaseMessage with Serializable {
    val messageId = JobAllowMultiCraftRequestSetMessage.messageId

    val enabled: Boolean

    def serialize(buf: Buffer) {
      buf.writeBoolean(enabled)
    }

  }

  protected def create(enabled: Boolean): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val enabled = buf.readBoolean()

    create(enabled)
  }

}

object JobAllowMultiCraftRequestSetMessage extends JobAllowMultiCraftRequestSetMessage {
  case class MessageImpl(enabled: Boolean) extends Message
  type Target = MessageImpl

  override def create(enabled: Boolean) = MessageImpl(enabled)

  def apply(enabled: Boolean) = create(enabled)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/JobAllowMultiCraftRequestSetMessage.java


// Generated on 05/08/2013 19:37:47
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class JobAllowMultiCraftRequestSetMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5749;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public boolean enabled;
    
    public JobAllowMultiCraftRequestSetMessage() { }
    
    public JobAllowMultiCraftRequestSetMessage(boolean enabled) {
        this.enabled = enabled;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeBoolean(enabled);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        enabled = buf.readBoolean();
    }
    
}
